tests/test_feedexport.py

```diff
@@ -1320,96 +1323,271 @@ def test_extend_kwargs(self):
+class FeedPostProcessedExportsTest(FeedExportTestBase):
+    __test__ = True
+    items = [
+        {'foo': 'bar1', 'baz': ''},
+        {'egg': 'bar2', 'baz': 'quux'},
+    ]
+    format_to_expected = {
+        'csv': b'foo,baz\r\nbar1,\r\n,quux\r\n',
+        'jsonlines': b'{"foo": "bar1", "baz": ""}\n{"egg": "bar2", "baz": "quux"}\n',
+        'xml': b'<?xml version="1.0" encoding="utf-8"?>\n<items>\n<item><foo>bar1</foo>'
+               b'<baz></baz></item>\n<item><egg>bar2</egg><baz>quux</baz></item>\n</items>',
+    }
+
     @defer.inlineCallbacks
-    def test_postprocessed_exports(self):
-        items = [
-            {'foo': 'bar1', 'baz': ''},
-            {'egg': 'bar2', 'baz': 'quux'},
-        ]
+    def run_and_export(self, spider_cls, settings):
+        """ Run spider with specified settings; return exported data. """
 
-        import gzip
-        import lzma
-        import bz2
-        format_to_decompressor = {
-            'csv': gzip,
-            'json': lzma,
-            'jsonlines': bz2,
-            'xml': gzip,
+        FEEDS = settings.get('FEEDS') or {}
+        settings['FEEDS'] = {
+            printf_escape(path_to_url(file_path)): feed_options
+            for file_path, feed_options in FEEDS.items()
         }
 
-        format_to_expected = {
-            'csv': b'foo,baz\r\nbar1,\r\n,quux\r\n',
-            'json': b'[\n{"foo": "bar1", "baz": ""},\n{"egg": "bar2", "baz": "quux"}\n]',
-            'jsonlines': b'{"foo": "bar1", "baz": ""}\n{"egg": "bar2", "baz": "quux"}\n',
-            'xml': b'XXXX<?xml version="1.0" encoding="utf-8"?>\nXXXX<itemsXXXX>XXXX\nX'
-                   b'XXX<itemXXXX>XXXX<fooXXXX>XXXXbar1XXXX</foo>XXXX<bazXXXX>XXXX</baz'
-                   b'>XXXX</item>XXXX\nXXXX<itemXXXX>XXXX<eggXXXX>XXXXbar2XXXX</egg>XXX'
-                   b'X<bazXXXX>XXXXquuxXXXX</baz>XXXX</item>XXXX\nXXXX</items>',
+        content = {}
+        try:
+            with MockServer() as s:
+                runner = CrawlerRunner(Settings(settings))
+                spider_cls.start_urls = [s.url('/')]
+                yield runner.crawl(spider_cls)
+
+            for file_path, feed_options in FEEDS.items():
+                if not os.path.exists(str(file_path)):
+                    continue
+
+                with open(str(file_path), 'rb') as f:
+                    content[feed_options['format']] = f.read()
+
+        finally:
+            for file_path in FEEDS.keys():
+                if not os.path.exists(str(file_path)):
+                    continue
+
+                os.remove(str(file_path))
+
+        return content
+
+    def get_gzip_compressed(self, data, compresslevel):
+        data_stream = BytesIO()
+        gzipf = gzip.GzipFile(fileobj=data_stream, filename="", mtime=0, compresslevel=compresslevel, mode="wb")
+        gzipf.write(data)
+        gzipf.close()
+        data_stream.seek(0)
+        return data_stream.read()
+
+    @defer.inlineCallbacks
+    def test_gzip_plugin(self):
```

Comment:
Instead of a single method per plugin, you could have several methods, each to test different aspects:
- Enabling the plugin results in gzip-compressed output
- A non-default value for a plugin option has the expected effect (1 test method per option)

I see no reason to test different output formats (i.e. csv, jsonlines, xml). Plugins should work the same with any, so to make things minimal you could use a single one.

What we should have is tests that verify that each of the Scrapy export plugins (for csv, jsonlines, etc.) work with the file-like interface of the plugin manager. In other words, tests that would have failed if the plugin manager was not inheriting from IOBase.